### PR TITLE
Add Simpler GLM CMake Config

### PR DESCRIPTION
**Pull request by Twinklebear**
This PR adds back in a CMake package config to allow finding GLM with CMake, with some improvements over the original one:

- It's no longer necessary to "build" glm to generate the CMake config file to use, this is now just included in the root of the repository. As a result the CMake config is also easy to package in the release archives as well, since it's as easy as including [`glmConfig.cmake`](https://github.com/Twinklebear/glm/blob/master/glmConfig.cmake) and [`glmConfig-version.cmake`](https://github.com/Twinklebear/glm/blob/master/glmConfig-version.cmake) in the zip file. An important note is that the version number tracked in these files must be updated for each release, since it's no longer generated during the CMake build.

- Added a test to verify the CMake package works on Travis and Appveyor (https://github.com/Twinklebear/glm/tree/master/test/cmake)

- Documentation in the manual on how to find GLM from CMake (https://github.com/Twinklebear/glm/blob/master/manual.md#section1_5)

How the new `glmConfig.cmake` and `glmConfig-version.cmake` work:

In CMake 3 the method for finding libraries has gotten a lot easier, when calling `find_package(<library>)` if the user passes `-D<library>_DIR=<path>` CMake will check that path for the library's package files (`<library>Config.cmake`). If a version is specified as `find_package(<library> 0.9.8)` it will use the `<library>Config-version.cmake` file to check if the version is suitable.

The library CMake packages can then export variables (2.x style) or targets (3.x style) which the user's code can use or link against. For header only libraries like glm we can use an `INTERFACE` target, which lets us set include directories, dependencies and compile options, but doesn't actually refer to any compiled libraries or code. The properties set on this target then get propagated to the user's code:

```cmake
cmake_minimum_required(VERSION 3.2 FATAL_ERROR)
cmake_policy(VERSION 3.2)

set(GLM_VERSION 0.9.9)
# Set the old GLM_INCLUDE_DIRS variable for backwards compatibility
set(GLM_INCLUDE_DIRS ${CMAKE_CURRENT_LIST_DIR})

# We define a new interface target glm::glm which is being imported from
# some existing code
add_library(glm::glm INTERFACE IMPORTED)
# If the user links our target against some library, the following directories
# will be added to their target's include directories
set_target_properties(glm::glm PROPERTIES
    INTERFACE_INCLUDE_DIRECTORIES ${CMAKE_CURRENT_LIST_DIR})
# Hide the glm_DIR parameter in the CMake UI
mark_as_advanced(glm_DIR)
```

Now in the user's CMakeLists:

```cmake
cmake_minimum_required(VERSION 3.2 FATAL_ERROR)
project(test_find_glm)

find_package(glm REQUIRED)

add_executable(test_find_glm test_find_glm.cpp)
# "Linking" the glm interface target will have the same effect as
# target_include_directories(test_find_glm ${GLM_INCLUDE_DIRS})
target_link_libraries(test_find_glm glm::glm)
```

### CODE REVIEW

**Comment by Twinklebear** [comment]
I'm not able to run the travis/appveyor stuff on my fork so I'll watch the PR for when those finish and fix the test if they fail

**Comment by Twinklebear** [comment]
It looks like the travis jobs that are failing are trying to use clang, but pick up gcc when building the glm tests. Then the new cmake test I added tries to use clang and fails. See: https://travis-ci.org/g-truc/glm/jobs/606156561#L186-L195 , https://travis-ci.org/g-truc/glm/jobs/606156554#L184-L195, https://travis-ci.org/g-truc/glm/jobs/606156555#L184-L195, https://travis-ci.org/g-truc/glm/jobs/606156557#L184-L195, https://travis-ci.org/g-truc/glm/jobs/606156559#L184-L194

**Comment by Twinklebear** [comment]
The appveyor tests look to be passing now, so once everything runs through I'll squash the CI fixing commits down

**Comment by PandarinDev** [comment]
Good job, looking forward to this getting merged back. :)

**Comment by Twinklebear** [comment]
The main difference of this and #968 is that this doesn't require users to "build" and install glm, which would give an easier to use path for finding GLM through CMake.

Also I should have time to maintain this, and I think the maintenance cost of this should be a lot lower than the previous CMake config.

**Comment by handsomematt** [comment]
This is much more favourable then #968 and integrates very easily in all my projects.

**Comment by mistydemeo** [comment]
It looks like this doesn't restore the `make install` target, so it doesn't resolve the core issue for many use cases. 😄 For that reason, I would prefer #968 over this branch.

**Comment by Twinklebear** [comment]
Could you explain your use case a bit more for the `make install` target? I'm curious because I have the opposite preference and always disliked having to "build" GLM through CMake just to generate the CMake config files. With this branch if you're building against a download of GLM you can just point CMake to the root of the repo with `-Dglm_DIR=<repo root>`. If you're not building through CMake you can set `-I<repo root>` to add the repo root to your include path. Do you use the `make install` to install it into your system include path?

**Comment by mistydemeo** [comment]
Yes, I work on a package manager that distributes `glm`. Without `make install`, we can't package and distribute it to our users.

**Comment by n8vm** [comment]
> Yes, I work on a package manager that distributes `glm`. Without `make install`, we can't package and distribute it to our users.

I also agree with @Twinklebear here. We shouldn’t need to run an install target just to get access to the cmake config file for a header only library. This requirement makes it impossible to submodule glm and point to the cloned config file, since it won’t exist until my end users run glm’s install target, which is counterintuitive to how other  header only libraries work.

An install target can still be used to package these config files, but the install target shouldn’t be required to generate and gain access to them.

**Comment by Twinklebear** [comment]
@mistydemeo gotcha, but since glm is header only do you just need the `make install` target to generate the install directories in the layout you need to distribute? Since glm is header only, it's already kind of in that layout, and it may be possible to install the `glm` dir in the include path, and I can move the cmake config stuff I put in the root dir I could put under the `cmake/glm` path and modify it to use a relative dir to find the include path (though I'm not sure how that works on installers/package managers). Does that sound like it would work?

**Comment by Groovounet** [comment]
Thanks @Twinklebear for your work. I'll follow your work for a future CMake PR hopefully. :D

**Comment by Twinklebear** [comment]
@Groovounet @mistydemeo I've moved the packaged CMake files into `cmake/glm`, so they can be copied into the user's `lib/` directory when installing so that they show up correctly when the package is installed. I think it should be fine now for the package to just take the `glm` dir and install that under the user's `include` path, and the `cmake` dir and install that in the `lib` path, and this will give the same installation behavior as before. Let me know how it looks

**Comment by Twinklebear** [comment]
I can squash down the commits for this before it's merged once everything looks good

**Comment by Groovounet** [comment]
Thanks for contributing !
Christophe

File: .gitignore [hunk 1]
```diff
@@ -56,3 +58,4 @@ build*
 /.vscode
 /CMakeSettings.json
 .DS_Store
+*.swp
```
[the remark this answers is not in the document]
Comment (Twinklebear) [reply .gitignore]: I added a gitignore rule for vim swap files, since I was using vim

**Comment by Twinklebear** [comment]
Thanks for merging it in @Groovounet ! I can fix up some of those comments you made as well (w/ the version numbering being repeated)

File: glmConfig-version.cmake [hunk 2]
```diff
@@ -0,0 +1,13 @@
+set(PACKAGE_VERSION 0.9.9)
```
[the remark this answers is not in the document]
Comment (Twinklebear) [reply glmConfig-version.cmake]: Yea, I moved them into the more standard cmake package path (`cmake/glm/`) which should also hopefully fix @mistydemeo's comment about package maintainers

**Pull request merged**